src/wallet/coin_selection.rs

```diff
@@ -138,6 +145,98 @@ impl CoinSelectionResult {
     }
 }
 
+/// Metric introduced to measure the performance of different coin selection algorithms.
+///
+/// This implementations considers waste as the sum of two values:
+/// * Timing cost
+/// * Creation cost
+/// > waste = timing_cost + creation_cost
+///
+/// **Timing cost** is the cost associated to the current fee rate and some long term fee rate used
+/// as a treshold to consolidate UTXOs.
+/// > timing_cost = txin_size * current_fee_rate - txin_size * long_term_fee_rate
+///
+/// Timing cost can be negative if the `current_fee_rate` is cheaper than the `long_term_fee_rate`,
+/// or zero if they are equal.
+///
+/// **Creation cost** is the cost associated to the surplus of coins besides the transaction amount
+/// and transaction fees. It can happen in the form of a change output or in the form of excess
+/// fees paid to the miner.
+///
+/// Change cost is derived from the cost of adding the extra output to the transaction and spending
+/// that output in the future.
+/// > change_cost = current_fee_rate * change_output_size + long_term_feerate * change_spend_size
+///
+/// Excess happens when there is not change, and the surplus of coins is spend as part of the fees
+/// to the miner:
+/// > excess = tx_total_value - tx_fees - target
+///
+/// Where _target_ is the desired amount to send.
+///
+/// Creation cost can be zero if there is a perfect match as result of the coin selection
+/// algorithm.
+///
+/// So, waste can be zero or negative if the creation cost is zero and the timing
+/// cost is less than or equal to zero.
```

Comment:
Nit: The waste score can also end up being negative if the creation cost is simply low rather than zero.

```suggestion
/// So, waste can be zero or negative if the creation cost is low or zero and the timing
/// cost is less than or equal to zero.
```

Reply:
What do you think of this?

> So, waste can be zero if creation and timing cost are zero. Or can be
> negative, if timing cost is negative and the creation cost is lower enough
> (less than the absolute value of timing cost).